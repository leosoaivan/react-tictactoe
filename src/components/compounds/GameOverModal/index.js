import React from 'react';
import PropTypes from 'prop-types';
import Modal from '../../atoms/Modal';

const GameOverModal = ({
  currentPlayerName,
  displayModal,
  gameResult,
  onClick,
}) => {
  let modalHeaderMessage = '';
  let modalBodyMessage = '';

  if (gameResult === 'win') {
    modalHeaderMessage = 'Congratulations';
    modalBodyMessage = `With strategy, cunning, and ruthelessness, ${currentPlayerName} has claimed absolute victory.`;
  } else {
    modalHeaderMessage = 'What a tired game';
    modalBodyMessage = 'Unfortunate. Sad. Demeaning to watch...a game where neither player is smart enough to outdo the other.';
  }

  return (
    <Modal
      displayModal={displayModal}
      modalHeader={modalHeaderMessage}
      modalBody={modalBodyMessage}
      onCloseModal={onClick}
      displayButton
    />
  );
};

GameOverModal.propTypes = {
  currentPlayerName: PropTypes.string,
  onClick: PropTypes.func,
  displayModal: PropTypes.bool,
  gameResult: PropTypes.string,
};

GameOverModal.defaultProps = {
  currentPlayerName: '',
  displayModal: false,
  gameResult: undefined,
  onClick: undefined,
};

export default GameOverModal;
