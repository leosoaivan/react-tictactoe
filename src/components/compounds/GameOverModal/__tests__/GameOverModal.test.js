import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import GameOverModal from '..';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('GameOverModal', () => {
  it('Renders an invisible GameOverModal by default', () => {
    const tree = renderer.create(
      <GameOverModal />,
    ).toJSON();

    expect(tree).toHaveStyleRule('display', 'none');
    expect(tree).toMatchSnapshot();
  });

  it('Renders with a winning message', () => {
    const wrapper = shallow(
      <GameOverModal
        displayModal
        gameResult="win"
      />,
    );

    expect(wrapper.prop('modalHeader')).toMatch('Congratulations');
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders with a losing message', () => {
    const wrapper = shallow(
      <GameOverModal
        displayModal
        gameResult="loss"
      />,
    );

    expect(wrapper.prop('modalHeader')).not.toMatch('Congratulations');
    expect(wrapper).toMatchSnapshot();
  });

  it('Initiates onClick callback', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(
      <GameOverModal
        displayModal
        onClick={onClickMock}
      />,
    );

    expect(onClickMock).toHaveBeenCalledTimes(0);
    wrapper.dive().find('CloseButton').simulate('click');
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
