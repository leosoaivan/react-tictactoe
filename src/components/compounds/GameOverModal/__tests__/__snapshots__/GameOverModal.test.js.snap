// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverModal Renders an invisible GameOverModal by default 1`] = `
.c0 {
  display: none;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0,0,0,0.6);
}

.c1 {
  position: fixed;
  background: white;
  width: 50%;
  height: 50%;
  top: 40%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  padding: 5%;
  text-align: center;
}

.c2 {
  font-size: 5vw;
  text-align: center margin:48px 0px;
}

.c3 {
  margin: auto;
  text-align: center;
}

.c4 {
  display: block;
  margin: 20px auto auto auto;
  height: 40px;
  width: 50%;
  background-color: green;
  color: white;
  font-weight: 600;
  font-size: 20px;
  border: none;
  border-radius: 2px;
}

.c4:hover {
  background-color: darkseagreen;
}

<div
  className="c0"
>
  <section
    className="c1"
  >
    <div
      className="c2"
    >
      What a tired game
    </div>
    <div
      className="c3"
    >
      Unfortunate. Sad. Demeaning to watch...a game where neither player is smart enough to outdo the other.
    </div>
    <button
      className="c4"
      onClick={null}
    >
      Try again, you'll feel better
    </button>
  </section>
</div>
`;

exports[`GameOverModal Renders with a losing message 1`] = `
<Modal
  displayButton={true}
  displayModal={true}
  modalBody="Unfortunate. Sad. Demeaning to watch...a game where neither player is smart enough to outdo the other."
  modalHeader="What a tired game"
  onCloseModal={null}
/>
`;

exports[`GameOverModal Renders with a winning message 1`] = `
<Modal
  displayButton={true}
  displayModal={true}
  modalBody="With strategy, cunning, and ruthelessness,  has claimed absolute victory."
  modalHeader="Congratulations"
  onCloseModal={null}
/>
`;
